corehq/motech/dhis2/static/dhis2/js/dataset_map_json.js

```diff
@@ -1,22 +1,44 @@
 hqDefine('dhis2/js/dataset_map_json', [
     'jquery',
+    'underscore',
     'knockout',
     'hqwebapp/js/initial_page_data',
     'hqwebapp/js/alert_user',
     'hqwebapp/js/base_ace',
+    'dhis2/js/json_syntax_parse',
 ], function (
     $,
+    _,
     ko,
     initialPageData,
     alertUser,
-    baseAce
+    baseAce,
+    jsonParse
 ) {
-
     var ViewModel = function (data) {
         var self = {};
         self.dataSetMap = ko.observable(JSON.stringify(data, null, 4));
+        self.errorMessage = ko.observable('');
+        self.isError = ko.computed(function () {
+            return self.errorMessage() !== '';
+        });
+
+        self.initMapConfigTemplate = function (elements) {
+            _.each(elements, function (element) {
+                _.each($(element).find('.jsonwidget'), baseAce.initObservableJsonWidget);
+            });
+        };
 
         self.submit = function (form) {
+            var editors = baseAce.returnEditors();
+            var value = editors[0].getValue();
+            try {
+                jsonParse.parseJson(value, null, 30);
+            } catch (error) {
+                self.errorMessage(error);
+                return self;
+            }
+            self.errorMessage('');
```

Comment:
I ended up not creating a function for doing this since I didn't want to define it 3 separate times and there was no obvious place to put a util function like that. I did add a small comment explaining what it does though.